### PR TITLE
Collapses Layout classes a bit further

### PR DESCRIPTION
**Pull request by tinwelint**
Since they no longer need to be split into unique/non-unique

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringLayout.java [hunk 1]
```diff
@@ -28,11 +28,11 @@
...
  * {@link Layout} for strings.
  */
-abstract class StringLayout extends SchemaLayout<StringSchemaKey>
+class StringLayout extends SchemaLayout<StringSchemaKey>
 {
-    StringLayout( long identifier, int majorVersion, int minorVersion )
+    StringLayout()
     {
-        super( identifier, majorVersion, minorVersion );
+        super( "USI", 0, 1 );
```
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringLayout.java:35]: "USI" -> static final field

**Review by burqen — CHANGES_REQUESTED** [review]
LGTM, just a small comment

File: community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringLayout.java (hunk 1 above)
Comment (tinwelint) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringLayout.java:35]: What would be the reason for that?

Comment (burqen) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringLayout.java:35]: Just to make it more explicit

Comment (burqen) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringLayout.java:35]: Never mind, this is overkill

**Review by burqen — APPROVED** [review]

**Pull request merged**